settings/src/components/svn-password.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { Button, Notice, Spinner } from '@wordpress/components';
+import { useCallback, useContext, useState } from '@wordpress/element';
+import { Icon, check, copySmall } from '@wordpress/icons';
+import { refreshRecord } from '../utilities/common';
+
+/**
+ * Internal dependencies
+ */
+import { GlobalContext } from '../script';
+
+/**
+ * Render the Email setting.
+ */
+export default function SVNPassword() {
+	const {
+		user: { userRecord },
+		setGlobalNotice,
+		setError,
+	} = useContext( GlobalContext );
+
+	const [ isGenerating, setGenerating ] = useState( false );
+	const [ generatedPassword, setGeneratedPassword ] = useState( '' );
+
+	// Generate a new SVN Password.
+	const handleGenerate = useCallback( async () => {
+		try {
+			setGenerating( true );
+
+			const response = await apiFetch( {
+				path: '/wporg-two-factor/1.0/generate-svn-password',
+				method: 'POST',
+				data: {
+					user_id: userRecord.record.id,
+				},
+			} );
+
+			setGeneratedPassword( response.svn_password );
+			setGenerating( false );
+
+			await refreshRecord( userRecord );
+		} catch ( apiFetchError ) {
+			setError( apiFetchError );
+		}
+	} );
+
+	const handleCopy = useCallback( () => {
+		try {
+			navigator.clipboard.writeText( generatedPassword );
+			setGlobalNotice( 'Copied to clipboard' );
+		} catch ( error ) {
+			setGlobalNotice( "Couldn't write to clipboard" );
```

Comment:
Is it possible to detect whether writing to the clipboard is possible before offering a button that won't work?

I can't come up with a neat way of phrasing it off the top of my head, but we could add words to the effect of "you're going to have to type it yourself" (ie offer a solution).

Reply:
On googling: seems like there's no reliable way to detect whether or not it will work.

Reply:
I thought I'd copied that from somewhere else in the project, but it looks like the only other place that has this is the backup codes that was recently reworked.

https://github.com/WordPress/wporg-two-factor/blob/trunk/settings/src/components/copy-to-clipboard-button.js 

We can probably do as that does and just not check..